enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/core/consensus/RaftMachine.java

```diff
@@ -124,28 +129,30 @@ public synchronized void startTimers()
         if ( !refuseToBecomeLeader )
         {
             lastElectionRenewalMillis = clock.millis();
-            electionTimer = renewableTimeoutService.create( Timeouts.ELECTION, electionTimeout, randomTimeoutRange(),
-                    renewing( this::electionTimeout ) );
-            heartbeatTimer = renewableTimeoutService.create( Timeouts.HEARTBEAT, heartbeatInterval, 0,
-                    renewing( () -> handle( new RaftMessages.Timeout.Heartbeat( myself ) ) ) );
+
+            electionTimer = timerService.create( Timeouts.ELECTION, Groups.raft, renewing( this::electionTimeout ) );
+            electionTimer.set( uniformRandomTimeout( electionTimeout, electionTimeout * 2, MILLISECONDS ) );
+
+            heartbeatTimer = timerService.create( Timeouts.HEARTBEAT, Groups.raft, renewing( () -> handle( new RaftMessages.Timeout.Heartbeat( myself ) ) ) );
+            heartbeatTimer.set( fixedTimeout( heartbeatInterval, MILLISECONDS ) );
         }
     }
 
     public synchronized void stopTimers()
     {
         if ( electionTimer != null )
         {
-            electionTimer.cancel();
+            electionTimer.cancel( true, true );
         }
         if ( heartbeatTimer != null )
         {
-            heartbeatTimer.cancel();
+            heartbeatTimer.cancel( true, true );
         }
     }
 
     private TimeoutHandler renewing( ThrowingAction<Exception> action )
```

Comment:
Could also be renamed to `TimerHandler`

Reply:
I think here the semantics are correct. A timer is like an egg timer and when it rings a timeout has occurred and thus you handle it with a timeout handler.